packages/synthchron-simulator/src/simulation.ts

```diff
@@ -96,34 +98,52 @@ export async function* simulateWithEngine<
     processEngine,
     processModel,
     simulationConfiguration,
-    simulationResults
+    simulationResults,
+    0
   )
   const randomGenerator = seedrandom(simulationConfiguration.randomSeed)
   // console.log(progress)
-  while (status.continue) {
+
+  for (
+    let step = 0;
+    step < simulationConfiguration.maximumTraces && status.continue;
```

Comment:
Aren't you checking the 'step < simulationConfiguration.maximumTraces' in the getSimulationProgress function?

Reply:
You're right, I implemented it here before merging the other change that has the "continue" attribute. Will clean it up